src/vm/prestub.cpp

```diff
@@ -1706,11 +1706,14 @@ PCODE MethodDesc::DoPrestub(MethodTable *pDispatchingMT)
     // for this method only then do we back-patch it.
     BOOL fCanBackpatchPrestub = TRUE;
 #ifdef FEATURE_TIERED_COMPILATION
+    TieredCompilationManager* pTieredCompilationManager = nullptr;
     BOOL fEligibleForTieredCompilation = IsEligibleForTieredCompilation();
+    BOOL fShouldPromoteToTier1 = FALSE;
     if (fEligibleForTieredCompilation)
     {
+        pTieredCompilationManager = GetAppDomain()->GetTieredCompilationManager();
         CallCounter * pCallCounter = GetCallCounter();
-        fCanBackpatchPrestub = pCallCounter->OnMethodCalled(this);
+        pCallCounter->OnMethodCalled(this, pTieredCompilationManager, &fCanBackpatchPrestub, &fShouldPromoteToTier1);
```

Comment:
I'm fine with checking this in for now, but can you add a comment that we plan to refactor it a bit as we were talking about?

Reply:
Sure I'll add some TODOs

Reply:
Moved async promote to tier 1 back to OnMethodCalled, I still need the extra boolean return since I don't want to record methods that have reached the call count threshold for tier 1. Added a TODO to PublishNativeCodeVersion.